homeassistant/components/modbus/sensor.py

```diff
@@ -340,3 +402,72 @@ def _update(self):
 
         self._available = True
         self.schedule_update_ha_state()
+
+
+class ModbusBitSensor(ModbusSensorBase):
+    """Modbus bit sensor."""
+
+    def __init__(
+        self,
+        hub,
+        name,
+        slave,
+        register,
+        register_type,
+        bit_number,
+        unit_of_measurement,
+        count,
+        device_class,
+        scan_interval,
+    ):
+        """Initialize the modbus bit sensor."""
+        super().__init__(
+            ModbusReadCache(hub),
```

Comment:
Only bit switch/sensor uses read cache as a wrapper. Regular sensors/switches would not use read cache.

Reply:
OK. Before you start doing changes, please make sure your branch is rebased against dev, so that this PR only contain your changes (also no merge commits), in order to facilitate review.

Regarding the cache, we are currently thinking about a different more general way. You instanciate an object of each  bit_sensor configured, but an alternative it is modbus.py to have 2 classes one with the real read and one with a virtual read (reusing the first read). That way you would instanciate the first bit_sensor with the real class (reading modbus) and the following with the second class. The reason we are thinking of this, is that we often see a configuration that read address, +1, +2 ..... and that like your bit_sensor is overload.